dbms/src/Storages/Transaction/TMTStorages.cpp

```diff
@@ -25,7 +27,11 @@ extern const int TIDB_TABLE_ALREADY_EXISTS;
 
 void ManagedStorages::put(ManageableStoragePtr storage)
 {
-    std::lock_guard lock(mutex);
+    // 用 unique_lock 的话重启的时候性能会有明显裂化
+    // std::lock_guard lock(mutex);
+    std::lock_guard lock(shared_mutex);
```

Comment:
It is an update operation, can we use `shared_mutex`?

Reply:
[std::unique_lock](https://github.com/pingcap/tiflash/pull/7437/unique_lock.html) lock(shared_mutex);